migrations/201906170001-Transaction-fixTaxAmount.js

```diff
@@ -0,0 +1,75 @@
+'use strict';
+
+import models, { sequelize, Op } from '../server/models';
+import Promise from 'bluebird';
+
+const fetchTransactions = async () => {
+  const query = { where: { taxAmount: { [Op.gt]: 0 } } };
+  if (process.env.LIMIT) {
+    query.limit = process.env.LIMIT;
+  }
+  return await models.Transaction.findAll(query);
+};
+
+const toNegative = n => (n > 0 ? -n : n);
+
+const fixTransaction = transaction => {
+  const updatedTransaction = { ...transaction.dataValues };
+  delete updatedTransaction.id;
+  updatedTransaction.taxAmount = toNegative(transaction.taxAmount);
+  if (transaction.type === 'CREDIT') {
```

Comment:
Why not doing all that in SQL queries? It would be more performant, more reliable (see comment above about using model in transactions) and the code would be shorter. Something like:

```sql
UPDATE Transactions
SET netAmountInCollectiveCurrency = netAmountInCollectiveCurrency + COALESCE(taxAmount, 0)
WHERE type = 'CREDIT'
```

```sql
UPDATE Transactions
SET
	amount = amount + COALESCE(taxAmount, 0),
	amountInHostCurrency = amountInHostCurrency - COALESCE(taxAmount, 0)
WHERE type = 'CREDIT'
```

Reply:
performance doesn't matter in those scripts that run in the background and only once.
So I favor readability and safety to avoid issues.

Hence also the use of postgres transactions to make sure I don't delete a transaction before making sure that the updated version is recorded.

Reply:
I personally find standard SQL queries way more readable than sequelize query language, but that's a matter of taste :)

Regarding safety however, it will always be safer to run a raw SQL query than to rely on sequelize and to convert the data in JS. And it supports transactions too.

Reply:
I actually agree with you 😅
It's only when I need to do data manipulation that I prefer to use javascript so that I can do console.log and verify the way the data looks before making any modification.